compiler/src/iree/compiler/Utils/EmbeddedDataDirectory.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_UTILS_EMBEDDEDDATADIRECTORY_H_
+#define IREE_COMPILER_UTILS_EMBEDDEDDATADIRECTORY_H_
+
+#include <mutex>
+#include "llvm/ADT/StringMap.h"
+
+namespace mlir::iree_compiler {
+
+// A string-to-StringRef map that acts as a virtual filesystem: the keys are
+// "filenames" and the values are file contents.
+class EmbeddedDataDirectory {
+public:
+  // A global singleton. Only use it for use cases that require it, such as when
+  // exporting data between parts of the compiler that can't directly link to
+  // each other (e.g. from a plugin to outside of the plugin).
+  //
+  // To avoid data races on this global singleton, all callers should also call
+  // getGlobalMutex and lock that around their accesses to the singleton.
+  static EmbeddedDataDirectory &getGlobal() {
+    static EmbeddedDataDirectory global;
+    return global;
+  }
+
+  // Mutex that callers of getGlobal should lock around accesses to the
+  // global singleton.
+  static std::mutex &getGlobalMutex() {
+    static std::mutex mutex;
+    return mutex;
+  }
+
+  // Add a new entry if it didn't already exist. Return `true` if it was added.
+  bool addFile(llvm::StringRef fileName, llvm::StringRef contents) {
+    auto [iter, success] = map.insert({fileName, contents});
```

Comment:
```suggestion
    auto [_iter, success] = map.insert({fileName, contents});
```
I've seen other code do this to indicate that the variable is discarded